### PR TITLE
[REFACT] Adds OIDC code as package

### PR DESCRIPTION
**Pull request by mdjunior**
This PR refact OIDC code as a package. This was done to improve the maintainability of the code and to insert tests into it.

After refactoring OIDC code, this PR adds FPM configuration for create RPM packages with gsh-agent. Other parts of GSH will be refactored to ensure a clean code at the next PR.

### CODE REVIEW

File: api/handlers/certificates.go [hunk 1]
```diff
@@ -89,21 +90,13 @@ func (h AppHandler) CertCreate(c echo.Context) error {
 	}
 
 	// Validating JWT before any other action
-	token, err := ValidateJWT(c, h.config)
+	ca := auth.OpenIDCAuth{}
+	username, err := ca.Authenticate(c, h.config)
 	if err != nil {
 		return c.JSON(http.StatusUnauthorized,
 			map[string]string{"result": "fail", "message": "Failed validating JWT", "details": err.Error()})
```
Comment (rafaveira3) [inline api/handlers/certificates.go]: Maybe a proper message here would be `"Authentication failed"`, as `ca.Authenticate` is now being used?

File: api/handlers/roles.go [hunk 2]
```diff
@@ -4,40 +4,24 @@ import (
...
 func (h AppHandler) GetRolesForMe(c echo.Context) error {
 	// Validates JWT token before any other action
-	token, err := ValidateJWT(c, h.config)
+	ca := auth.OpenIDCAuth{}
+	username, err := ca.Authenticate(c, h.config)
 	if err != nil {
 		return c.JSON(http.StatusUnauthorized,
 			map[string]string{"result": "fail", "message": "Failed validating JWT", "details": err.Error()})
 	}
-	field := h.config.GetString("oidc_claim")
```
Comment (rafaveira3) [inline api/handlers/roles.go]: Why did you remove this part of the code? The deletion is made several times in this file but I could get it yet. 🙃

File: Makefile [hunk 3]
```diff
@@ -82,4 +82,8 @@ compact:
 			echo "-> Generate tar for $${GOOS}/$${GOARCH}..."; \
 			tar cfz dist/gsh-$${GOOS}-$${GOARCH}.tar.gz README.md LICENSE -C dist/$${GOOS}/$${GOARCH} .; \
 		done; \
-	done
\ No newline at end of file
+	done
```
Comment (rafaveira3) [inline Makefile]: I also missed a function to test this new `_test.go` file. 

Something that would eventually run this: ```go test -v ./...```

File: api/handlers/roles.go (hunk 2 above)
Comment (mdjunior) [reply api/handlers/roles.go]: At the previous code, the function `ValidateJWT` returns the IDToken (and error). Using the IDToken, the next code gets the field with the username.
```
	username, err := getField(&token, field)	
	if err != nil {	
		return c.JSON(http.StatusInternalServerError,	
			map[string]string{"result": "fail", "message": "The field declared in oidc_claim doesn't exist", "details": err.Error()})	
	}
```
The new code `ca.Authenticate` returns the username (and error) and set as a context variable the JTI. This approach doesn't need the previous logic.

File: api/auth/oidc.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+	if keys != nil {
+		return nil
+	}
+
+	// Get JWT keys from OICD Provider (keys are not defined at configuration)
+	client := &http.Client{
+		Timeout: time.Duration(10) * time.Second,
+	}
+	keyURL := config.GetString("oidc_base_url") + "/" + config.GetString("oidc_realm") + "/protocol/openid-connect/certs"
+	req, _ := http.NewRequest("GET", keyURL, nil)
```
[the remark this answers is not in the document]
Comment (mdjunior) [reply api/auth/oidc.go]: Done!

File: api/handlers/certificates.go (hunk 1 above)
Comment (mdjunior) [reply api/handlers/certificates.go]: Done!

File: Makefile (hunk 3 above)
Comment (mdjunior) [reply Makefile]: Great, I make it at #52.

File: api/handlers/roles.go (hunk 2 above)
Comment (rafaveira3) [reply api/handlers/roles.go]: Nice! I got your point! Good refactoring!
